### ISSUE TITLE
Openhab Android app compatability with AndroidAuto

### ISSUE THREAD

**Issue by lb4207** [openhab/openhab-android#3041]
It appears the Openhab Android App is not Android Auto compatible as it doesn't appear in available apps in the Android Auto Launcher.

I have a garage door opener configured in Openhab and I can open my garage door from the app on my phone but now that I have a car with Android Auto I would like to be able open it from the console in the car or even program a steering wheel button to open the door.

**Comment by mueller-ma** [comment]
Seems like a valid feature request, but will probably not fulfilled soon. PRs are welcomed, but thoughts on the implementation should be shared in this issue before.

**Comment by felixschndr** [comment]
An alternative is to use Google Assistant. That's how I control my smart home devices while driving. Since my steering wheel has a button for calling Google Assistant this is pretty convenient

**Comment by maniac103** [comment]
Since I recently got a car with AA capability, I had an initial look at what's needed to support it, and got discouraged pretty soon:
- AA introduction is [here](https://developer.android.com/training/cars), it looks like we'd fit into the POI category well
- As POI app, we'd need to use their template system as documented [here](https://developer.android.com/training/cars/apps)
- The available template parts are documented [here](https://developers.google.com/cars/design/create-apps/apps-for-drivers/overview) - we'd probably want to use a list template
- List items can have [these parts](https://developers.google.com/cars/design/create-apps/apps-for-drivers/components/row#row-options) - what's interesting is what's missing: we neither can use buttons nor sliders, let alone maps, WebView, videos or the like
- so for sitemaps, only Text, Switch (but not switch-with-mappings) and maybe Selection (via multiple radio buttons) would be possible AFAICT, no Setpoint, Slider or anything more advanced 

Given that, I'm not sure whether it's actually useful to pursue this? Personally I wanted to look into implementing this to get access to my garden gate and garage door opener, but how could one control such things without buttons?

**Comment by felixschndr** [comment]
> Given that, I'm not sure whether it's actually useful to pursue this?

I agree. I mean a full fletched openhab sitemap in the car would be nice to have but is not necessary at all (imo). When I want to control something from _not at home_ (which is not often) I usually just use Google assistant with voice commands or just use the app on the phone quickly.
For something like a garage door opener it would be cool, but there might be different approaches to that as well (Homescreen shortcut, shortcut in quick settings, Google assistant voice commands, open/close depending on location/wifi of device).

Moreover, opening/closing is one of the (imo) very few use cases where AA integration would be useful. Maybe turning all lights off or something... what do the others think about this? Do you have an idea/usecase where an AA integration this limited could be worth implementing?

**Comment by mueller-ma** [comment]
> AA introduction is [here](https://developer.android.com/training/cars), it looks like we'd fit into the POI category well

What about the IoT category? Description of that is "Examples include controlling the state of certain devices, such as opening a garage door, flipping home light switches, or enabling home security.", so exactly what openHAB can be used for.

> The available template parts are documented [here](https://developers.google.com/cars/design/create-apps/apps-for-drivers/overview) - we'd probably want to use a list template

The grid template should work as well. Here's also a screenshot from a smart home app: https://developers.google.com/cars/design/create-apps/apps-for-drivers/templates/grid-template
One could configure Sitemaps that only have switches without mapping (or a maximum of 2 mappings). That's a quite limited set of widgets, but should be enough for some basic tasks, like opening the garage door.

**Comment by maniac103** [comment]
> What about the IoT category?

Oops, that was a typo/thinko. I actually meant to say 'we'd fit into the IoT category well'.

Another thought for section switches and selections: present them as text only first (in normal list), and upon click, present a new list (or grid) showing the 'buttons' (switches with mappings) or selectable values (selection). Setpoint could probably also be handled similarly.

**Comment by maniac103** [comment]
I've pushed a protype for an AA implementation [here](https://github.com/maniac103/openhab.android/tree/car). Text and frames work; what's left to figure out is how to implement the possible UI elements. Toggle switches can be rendered via `Row.Builder.setToggle()`, switch-with-mappings and selections could likely get their own list screen with radio buttons (via `ItemList.Builder.setOnSelectedListener()`.

**Comment by fabgio** [comment]
Thanks 👍
